nemo_skills/evaluation/evaluator/mcq.py

```diff
@@ -24,11 +25,27 @@
 
 
 def eval_mcq(cfg):
+    def extract_letter(text):
+        # extract prediction from boxed{}
+        parsed = extract_answer(text)
+        if parsed is not None and len(parsed) != 1:
+            match = re.findall(r"\b[A-J]\b(?!.*\b[A-J]\b)", parsed, re.DOTALL)
+            if len(match) > 0:
+                parsed = match[-1].strip()
+
+        # adapted from https://artificialanalysis.ai/methodology/intelligence-benchmarking#intelligence-index-evaluation-suite-overview
+        if parsed is None:
+            match = re.findall(r"(?i)[\*\_]{0,2}Answer[\*\_]{0,2}\s*:[\s\*\_]{0,2}\s*([A-Z])(?![a-zA-Z0-9])", text)
```

Comment:
Why do we look here for A-Z answers, and in the previous condition for A-J only?